src/Tizen.Network.Nsd/Tizen.Network.Nsd/DnssdService.cs

```diff
@@ -250,18 +307,58 @@ public void AddTXTRecord(string key, string value)
             {
                 Log.Error(Globals.LogTag, "Failed to add the TXT record, Error: " + (DnssdError)ret);
                 NsdErrorFactory.ThrowDnssdException(ret);
-            }
-
-            byte[] txtValue;
-            ushort txtLength;
-            GetTxtRecord(out txtLength, out txtValue);
+            }            
+        }
 
-            ret = Interop.Nsd.Dnssd.SetRecord(_serviceHandle, _dnsRecordtype, txtLength, txtValue);
+        /// <summary>
+        /// Set the record.
+        /// </summary>
+        /// <remarks>
+        /// TXT record should be added after registering the local service using RegisterService().
+        /// </remarks>
+        /// <returns>on success returns zero else returns error number.</returns>
+        /// <since_tizen> 9 </since_tizen>
+        /// <param name="type">type of the record to set</param>
+        /// <param name="txtValue">The value of the TXT record</param>
+        /// <feature>http://tizen.org/feature/network.service_discovery.dnssd</feature>
+        /// <exception cref="NotSupportedException">Thrown when DNS-SD is not supported.</exception>
+        /// <exception cref="ArgumentException">Thrown when the value of the key is null.</exception>
+        /// <exception cref="InvalidOperationException">Thrown when any other error occurred.</exception>
+        public int SetRecord(ushort type, byte[] txtValue)
```

Comment:
❌ DO NOT return error codes.

Exceptions are the primary means of reporting errors in frameworks.

https://docs.microsoft.com/en-us/dotnet/standard/design-guidelines/exception-throwing

Reply:
Why not use enum type on `type` parameter?
Which value could be use in here?